frontend/src/entities/batch-translate/create/batch-translate-create.tsx

```diff
@@ -263,19 +193,24 @@ export function BatchTranslateCreate () {
                 })
             );
 
-            if (response.type.endsWith('/fulfilled')) {
+            if (isFulfilled(response)) {
                 notificationService.generateNotification(
                     `Successfully created translation job ${state.form.JobName}.`,
                     'success'
                 );
-                const newDataset = {
-                    name: state.form.OutputDataset.Name,
-                    description: `Dataset created as part of the Batch Translate job: ${state.form.JobName}`,
-                    type: state.form.OutputDataset.Type,
-                    scope: determineScope(state.form.OutputDataset.Type, projectName, userName!)
-                } as IDataset;
+                
+                const dataset = datasetFromS3Uri(state.form.OutputDataConfig.S3Uri);
+                if (dataset) {
+                    const newDataset = {
+                        name: dataset.name,
+                        description: `Dataset created as part of the Batch Translate job: ${state.form.JobName}`,
```

Comment:
Adding this description feels like it should be something handled in the `createDatasetHandleAlreadyExists` function. Why is an entirely new object being created here when only a description is being appended?

Reply:
I agree this can be cleaned up. Originally `datasetFromS3Uri` returned an untyped object. I gave it the `DatasetContext` type to clean it up a little but it evolved to be close enough to `IDataset` I can redefine it using IDataset. This allows the code to be cleaned up a lot.

```
                const dataset = datasetFromS3Uri(state.form.OutputDataConfig.S3Uri);
                if (dataset) {
                    dataset.description = `Dataset created as part of the Batch Translate job: ${state.form.JobName}`;    
                    createDatasetHandleAlreadyExists(dataset);
                }
```